server/s3_put_object_action.cc

```diff
@@ -152,7 +152,7 @@ void S3PutObjectAction::validate_put_request() {
              request->get_user_metadata_size() > MAX_USER_METADATA_SIZE) {
 
     s3_put_action_state = S3PutObjectActionState::validationFailed;
-    set_s3_error("BadRequest");
+    set_s3_error("MetadataTooLarge");
```

Comment:
Can you please verify that in case if the header length crosses 8K for PUT request appropriate error ( MetadataTooLarge) is returned, currently its BadRequest
s3_put_chunk_upload_object_action.cc, s3_put_multiobject_action.cc, s3_put_object_action.cc

Reply:
Fixed.

Reply:
fixed